In [1]:
import pandas as pd
df = pd.read_csv("C:\\Users\\sasip\\OneDrive\\Desktop\\data-cleaning-pipelines\\data\\raw\\SampleSuperstore.csv")
df.head()

,Ship Mode,Segment,Country,City,State,Postal Code,Region,Category,Sub-Category,Sales,Quantity,Discount,Profit
0,Second Class,Consumer,United States,Henderson,Kentucky,42420,South,Furniture,Bookcases,261.9600,2,0.00,41.9136
1,Second Class,Consumer,United States,Henderson,Kentucky,42420,South,Furniture,Chairs,731.9400,3,0.00,219.5820
2,Second Class,Corporate,United States,Los Angeles,California,90036,West,Office Supplies,Labels,14.6200,2,0.00,6.8714
3,Standard Class,Consumer,United States,Fort Lauderdale,Florida,33311,South,Furniture,Tables,957.5775,5,0.45,-383.0310
4,Standard Class,Consumer,United States,Fort Lauderdale,Florida,33311,South,Office Supplies,Storage,22.3680,2,0.20,2.5164


In [2]:
df = pd.read_csv("../data/raw/SampleSuperstore.csv")

In [3]:
df.shape

(9994, 13)

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9994 non-null   str    
 1   Segment       9994 non-null   str    
 2   Country       9994 non-null   str    
 3   City          9994 non-null   str    
 4   State         9994 non-null   str    
 5   Postal Code   9994 non-null   int64  
 6   Region        9994 non-null   str    
 7   Category      9994 non-null   str    
 8   Sub-Category  9994 non-null   str    
 9   Sales         9994 non-null   float64
 10  Quantity      9994 non-null   int64  
 11  Discount      9994 non-null   float64
 12  Profit        9994 non-null   float64
dtypes: float64(3), int64(2), str(8)
memory usage: 1015.1 KB


In [5]:
df.isnull().sum()

Ship Mode       0
Segment         0
Country         0
City            0
State           0
Postal Code     0
Region          0
Category        0
Sub-Category    0
Sales           0
Quantity        0
Discount        0
Profit          0
dtype: int64

In [6]:
df.duplicated().sum()

np.int64(17)

In [7]:
df.describe()

,Postal Code,Sales,Quantity,Discount,Profit
count,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,55190.379428,229.858001,3.789574,0.156203,28.656896
std,32063.693350,623.245101,2.225110,0.206452,234.260108
min,1040.000000,0.444000,1.000000,0.000000,-6599.978000
25%,23223.000000,17.280000,2.000000,0.000000,1.728750
50%,56430.500000,54.490000,3.000000,0.200000,8.666500
75%,90008.000000,209.940000,5.000000,0.200000,29.364000
max,99301.000000,22638.480000,14.000000,0.800000,8399.976000


In [8]:
df.columns


Index(['Ship Mode', 'Segment', 'Country', 'City', 'State', 'Postal Code',
       'Region', 'Category', 'Sub-Category', 'Sales', 'Quantity', 'Discount',
       'Profit'],
      dtype='str')

### Observations

- Dataset contains 9994 rows and 13 columns
- No missing values found
- 17 duplicate rows present
- Column names contain spaces and inconsistent formatting
- Postal Code is numeric but represents categorical data
- Profit column contains negative values (business loss)
- Sales column has extreme high values (potential outliers)

In [1]:
import sys
print(sys.executable)

C:\Users\sasip\OneDrive\Desktop\data-cleaning-pipelines\venv\Scripts\python.exe


In [3]:
import sys
sys.path.append("../src")

from cleaning import clean_column_names, remove_duplicates, fix_postal_code

In [4]:
df = clean_column_names(df)
df = remove_duplicates(df)
df = fix_postal_code(df)

df.head()

NameError: name 'df' is not defined

In [5]:
import matplotlib.pyplot as plt

#boxplot---
plt.boxplot(df['sales'])
plt.title("Sales Outliers")
plt.show()

ModuleNotFoundError: No module named 'matplotlib'

In [6]:
import matplotlib.pyplot as plt

# Boxplot for sales
plt.boxplot(df['sales'])
plt.title("Sales Outliers")
plt.show()

ModuleNotFoundError: No module named 'matplotlib'